### ISSUE TITLE
Use module.version_numified, even for exact version matches

### ISSUE THREAD

**Issue by SineSwiper** [miyagawa/cpanminus#533]
This is a continuation of https://github.com/Perl-Toolchain-Gang/CPAN-Meta-Requirements/issues/32.

Within [version_to_query](https://github.com/miyagawa/cpanminus/blob/devel/lib/App/cpanminus/script.pm#L499), it skips numified version checking on exact matches.  This breaks alpha version matching, inexact numeric matches like `1.013 == v1.13`, and pretty much anything that has been reworked with `version`.  Here's the proposed change:

        return {
            term => { 'module.version_numified' => $self->numify_ver_metacpan($req) },
        };

**Comment by miyagawa** [comment]
Have you tested it?

**Comment by SineSwiper** [comment]
Sorry, getting back to this.  I've got something that should work, including a versions upgrade, and a xt unit test.  But, it doesn't because of what appears to be a legacy shim in place for MetaCPAN.

    use strict;
    use warnings;
    use LWP::UserAgent ();

    my $ua = LWP::UserAgent->new;
    $ua->timeout(10);
    $ua->agent('cpanminus/1.7043 perl/5.018002');

    my $response = $ua->get('http://api.metacpan.org/v0/file/_search?source=%7B%22query%22%3A%7B%22filtered%22%3A%7B%22query%22%3A%7B%22nested%22%3A%7B%22query%22%3A%7B%22custom_score%22%3A%7B%22metacpan_script%22%3A%22score_version_numified%22%2C%22query%22%3A%7B%22constant_score%22%3A%7B%22filter%22%3A%7B%22and%22%3A%5B%7B%22term%22%3A%7B%22module.authorized%22%3Atrue%7D%7D%2C%7B%22term%22%3A%7B%22module.indexed%22%3Atrue%7D%7D%2C%7B%22term%22%3A%7B%22module.name%22%3A%22Perl%3A%3AVersion%22%7D%7D%2C%7B%22term%22%3A%7B%22module.version_numified%22%3A%221.013030%22%7D%7D%5D%7D%7D%7D%7D%7D%2C%22score_mode%22%3A%22max%22%2C%22path%22%3A%22module%22%7D%7D%7D%7D%2C%22fields%22%3A%5B%22date%22%2C%22release%22%2C%22author%22%2C%22module%22%2C%22status%22%5D%7D');

    print $response->decoded_content;

Run that, and it errors.  Comment out the agent line, and it works.

Is the shim in place because cpanm is still using v0 of the API?  This sounds like interference from #520 and/or #522.

**Comment by miyagawa** [comment]
yeah sounds related to the specific shim for cpanm User-Agent. @haarg

**Comment by haarg** [comment]
I've fixed the shim itself, which was a trivial change.  However, the /download_url/ endpoint doesn't handle this correctly at this time.  For exact version specifications, it checks against module.version, rather than checking the numified versions.  This should probably be changed.

**Comment by haarg** [comment]
This should be fixed now.

**Comment by haarg** [comment]
The shim will now include information about how it parsed the incoming query in its responses.
